### ISSUE TITLE
Warnings when enabling WP Rocket on PHP8

### ISSUE THREAD

**Issue by piotrbak** [wp-media/wp-rocket#3249]
**Describe the bug**
When enabling WP Rocket for the first time after installation the following errors are present:
```
[27-Oct-2020 15:51:19 UTC] PHP Warning:  Private methods cannot be final as they are never overridden by other classes in /home/customer/www/php8.rocketlabsqa.dev/public_html/wp-content/plugins/wp-rocket/inc/classes/Buffer/class-cache.php on line 425
[27-Oct-2020 15:51:19 UTC] PHP Warning:  Private methods cannot be final as they are never overridden by other classes in /home/customer/www/php8.rocketlabsqa.dev/public_html/wp-content/plugins/wp-rocket/inc/classes/traits/trait-memoize.php on line 87
```

**To Reproduce**
1. Install WP Rocket
2. Activate it
3. Refresh the admin page
4. Check the error log

**Expected behavior**
WP Rocket working on PHP8 without problems.

**Additional context**
Errors in snippi:
https://snippi.com/s/ohe1juv

https://github.com/php/php-src/blob/PHP-8.0/UPGRADING
`Applying the final modifier on a private method will now produce a warning unless that method is the constructor.
`

**Backlog Grooming (for WP Media dev team use only)**
- [x] Reproduce the problem
- [x] Identify the root cause
- [x] Scope a solution
- [x] Estimate the effort

**Comment by hellofromtonya** [comment]
@arunbasillal @Tabrisrp We should fix this before 5.6 comes out. This particular PHP 8 warning is fixed in WP 5.6 in [Trac Ticket 50897](https://core.trac.wordpress.org/ticket/50897).

**Comment by Tabrisrp** [comment]
### Scope a solution
- Remove the `final` keyword from the `define_donotoptimize_true()` method in `inc/classes/Buffer/class-cache.php`
- Remove the `final` keyword from the `get_memoize_args_hash()` method in `in/classes/traits/trait-memoize.php`

### Estimate the effort
Effort `[XS]`
